Editor/Data/Type/DefaultObservableListDrawer.cs

```diff
@@ -0,0 +1,109 @@
+        protected const float objectPickerPadding = 4f;
+        protected readonly GUIStyle buttonStyle = new GUIStyle("RL Footer");
+        protected readonly GUIContent objectPickerIcon = EditorGUIUtility.TrIconContent("In ObjectField");
+        protected readonly GUIContent buttonIconAdd = EditorGUIUtility.TrIconContent("Toolbar Plus", "Add element to list");
+        protected readonly GUIContent buttonIconRemove = EditorGUIUtility.TrIconContent("Toolbar Minus", "Remove last element from list");
+
+        public override void OnGUI(Rect position, SerializedProperty property, GUIContent label)
+        {
+            {
+                label.tooltip = EditorHelper.GetTooltipAttribute(fieldInfo)?.tooltip ?? string.Empty;
+                EditorGUILayout.PropertyField(property, label, true);
+                object propertyObject = fieldInfo.GetValue(property.serializedObject.targetObject);
+                if (propertyObject == null)
+                {
+                    return;
+                }
+
+                buttonStyle.overflow = new RectOffset(2, 0, 2, 0);
+                buttonStyle.fixedHeight = buttonIconAdd.image.height * 1.01f;
+                buttonStyle.fixedWidth = objectPickerIcon.image.width * 1.5f;
+
+                using (new EditorGUI.IndentLevelScope())
+                {
+                    property.isExpanded = EditorGUILayout.Foldout(property.isExpanded, string.Format(collectionLabel, label.text), true);
+                    if (property.isExpanded)
+                    {
+                        using (new EditorGUI.IndentLevelScope())
+                        {
+                            Type listType = propertyObject.GetType();
+                            dynamic list = propertyObject;
+                            bool listIsEmpty = list.NonSubscribableElements.Count == 0;
+
+                            if (listIsEmpty)
+                            {
+                                using (new EditorGUI.IndentLevelScope(-2))
+                                {
+                                    EditorGUILayout.BeginHorizontal();
+                                    EditorGUILayout.PrefixLabel(" ");
+                                    EditorGUILayout.HelpBox(collectionEmptyLabel, MessageType.None);
+                                    GUILayout.Space(objectPickerIcon.image.width - objectPickerPadding + 1);
+                                    EditorGUILayout.EndHorizontal();
+                                }
+                            }
+                            else
+                            {
+                                for (int index = 0; index < list.NonSubscribableElements.Count; index++)
+                                {
+                                    var currentElement = list.NonSubscribableElements[index];
+                                    Type elementType = currentElement == null ? list.ElementType : currentElement.GetType();
```

Comment:
```suggestion
                                    Type elementType = currentElement == null ? GetElementType() : currentElement.GetType();
```
Here's the method implementation, to be put into the class as an instance method:
```c#
        protected virtual Type GetElementType()
        {
            Type type = fieldInfo.FieldType;
            while (type.BaseType != null)
            {
                type = type.BaseType;
                if (type.IsGenericType && type.GetGenericTypeDefinition() == typeof(ObservableList<,>))
                {
                    return type.GenericTypeArguments[0];
                }
            }

            throw new InvalidOperationException("The element type was not found.");
        }
```